src/bootstrap/src/utils/helpers.rs

```diff
@@ -473,6 +473,14 @@ pub fn linker_flags(
                 // cfg(bootstrap) - remove after updating bootstrap compiler (#137498)
                 if stage == 0 && target.is_windows() {
                     args.push("-Clink-arg=-fuse-ld=lld".to_string());
+                } else if target.starts_with("wasm") {
+                    if target.ends_with("unknown") || target.ends_with("none") {
+                        args.push("-Clinker-flavor=wasm-lld".to_string());
+                    } else if target.ends_with("emscripten") {
+                        args.push("-Clinker-flavor=em-cc".to_string());
+                    } else {
+                        args.push("-Clinker-flavor=wasm-lld-cc".to_string());
+                    }
```

Comment:
Alternatively you could omit the `-Clinker-flavor` arg entirely on wasm given that wasm already uses lld by default (there is no other wasm linker that I know of).

Reply:
We should use `-Zlinker-features` instead of the flavor, I messed this up. I fixed it in https://github.com/rust-lang/rust/pull/139378.